Include/Rover/Sample.hpp

```diff
@@ -89,6 +99,25 @@ namespace Rover {
   constexpr std::size_t arguments_size(const Arguments& args) {
     return ArgumentVisitor<Arguments>::size(args);
   }
+  
+  template<typename R, typename... A>
+  std::ostream& operator <<(std::ostream& stream, const Sample<R, A...>&
+      sample) {
+    stream << sample.m_result;
+    std::apply([&](const auto&... arg) {
+      ((stream << ',' << arg), ...);
+    }, sample.m_arguments);
+    return stream;
+  }
+
+  template<typename R, typename... A>
+  std::istream& operator >>(std::istream& stream, Sample<R, A...>& sample) {
+    stream >> sample.m_result;
+    std::apply([&](auto&... arg) {
```

Comment:
After each value, test that the stream is valid. If an error occurs you will need to immediately abort streaming.